Add title, frontend_title to membership type

mattwire opened the pull request:
Overview
----------------------------------------
https://lab.civicrm.org/dev/core/-/issues/4627

See https://github.com/civicrm/civicrm-core/pull/28182 and https://github.com/civicrm/civicrm-core/pull/34153

Re-work of original PR from @eileenmcnaughton 

Before
----------------------------------------
Membership Types have a single "name" field which is localizable. Means they have to be referenced by ID which changes on different sites..

After
----------------------------------------
Membership Types have a non-localizable "name" field which can be unique across sites. Localizable title, frontend_title added.

Technical Details
----------------------------------------


Comments
----------------------------------------
Still needs a few changes to display "title/frontend_title" instead of "name" on frontend forms - maybe in follow-up PRs?

mattwire committed:
MembershipType - Test fixes

colemanw commented on schema/Member/MembershipType.entityType.php:
```diff
@@ -56,12 +63,37 @@
       'title' => ts('Membership Type'),
       'sql_type' => 'varchar(128)',
-      'input_type' => 'Text',
+      'input_type' => NULL,
```
Don't think this change is needed.

colemanw committed:
MembershipType - Show titles instead of name on admin page/form

mattwire committed:
Remove legacy MembershipPayment check from test

colemanw committed:
MembershipType - Add a new schemaStructure file

Without this, the multilingual upgrader crashes.
But we really need to get away from our upgrader's dependence on these, as they are no longer generated by EFv2.

mattwire committed:
MembershipType - Add title, frontend_title; de-localize name

mattwire committed:
Add upgrader for Membership title/frontend_title

mattwire committed:
Use frontend_title for MembershipType on Contribution Pages, use title for MembershipType in backend

mattwire commented:
> @mattwire that's cool. We could merge this now but it still feels half finished because e.g. on a contribution form with selectable membership type it's not actually showing the frontend title.

ok. I just added another commit which switches various places in the UI (including contribution pages) to use frontend_title/title. I'm sure there will be more places but I think I got most of the obvious ones. Also caught an issue with creating new membershiptypes being broken - now fixed by switching the create API call to writeRecord.

Commit:
regenerate civicrm_generated
